lib/runtime-c-api/wasmer.h

```diff
@@ -504,6 +517,36 @@ wasmer_result_t wasmer_module_instantiate(const wasmer_module_t *module,
                                           wasmer_import_t *imports,
                                           int imports_len);
 
+/**
+ * Serialize the given Module.
+ * The caller owns the object and should call `wasmer_serialized_module_destroy` to free it.
+ * Returns `wasmer_result_t::WASMER_OK` upon success.
+ * Returns `wasmer_result_t::WASMER_ERROR` upon failure. Use `wasmer_last_error_length`
+ * and `wasmer_last_error_message` to get an error message.
+ */
+wasmer_result_t wasmer_module_serialize(wasmer_serialized_module_t **serialized_module,
+                                        const wasmer_module_t *module);
+
+/**
+ * Get bytes of the serialized module.
+ */
+wasmer_byte_array wasmer_serialized_module_bytes(const wasmer_serialized_module_t *serialized_module);
+
+/**
+ * Frees memory for the given serialized Module.
+ */
+void wasmer_serialized_module_destroy(wasmer_serialized_module_t *serialized_module);
+
+/**
+ * Transform a sequence of bytes into a serialized module.
+ * The caller owns the object and should call `wasmer_serialized_module_destroy` to free it.
+ * Returns `wasmer_result_t::WASMER_OK` upon success.
+ * Returns `wasmer_result_t::WASMER_ERROR` upon failure. Use `wasmer_last_error_length`
+ * and `wasmer_last_error_message` to get an error message.
+ */
+wasmer_result_t wasmer_serialized_module_from_bytes(wasmer_serialized_module_t **serialized_module,
+                                                    const wasmer_byte_array *serialized_module_bytes);
```

Comment:
It may also be convenient to go directly to `wasmer_module_t` from bytes.

Reply:
As discussed in private, let's see if we need it in the future.